saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/SavedSitesRepository.kt

```diff
@@ -336,8 +336,13 @@ class RealSavedSitesRepository(
             savedSitesRelationsDao.relationByEntityId(entity.entityId)?.let { relation ->
                 if (relation.folderId != bookmark.parentId) {
                     // bookmark moved to another folder
+                    val isFavorite = getFavorite(bookmark.url) != null
                     savedSitesRelationsDao.deleteRelationByEntity(bookmark.id)
                     savedSitesRelationsDao.insert(Relation(folderId = bookmark.parentId, entityId = relation.entityId))
+                    if (isFavorite) {
+                        // favorite relations must be restored too
+                        savedSitesRelationsDao.insert(Relation(folderId = SavedSitesNames.FAVORITES_ROOT, entityId = relation.entityId))
```

Comment:
This will move the favorite to the end of the list when moving the bookmark to a folder.